File: documentation/user/en/solve/examples/filtering-products-in-category/price-filter.cs
EvitaResponse<ISealedEntity> entities = evita.QueryCatalog(
	"evita",
	session => session.QuerySealedEntity(
        Query(
        	Collection("Product"),
        	FilterBy(
        		EntityLocaleEquals(CultureInfo.GetCultureInfo("en")),
        		HierarchyWithin(
        			"categories",
        			AttributeEquals("url", "/en/smartwatches")
        		),
        		PriceInPriceLists("basic"),
        		PriceInCurrency(new Currency("EUR")),
        		PriceValidInNow(),
        		UserFilter(
        			PriceBetween(50m, 400m)
        		)
        	),
        	Require(
        		PriceHistogram(10, Standard)
        	)
        )
	)
);

File: documentation/user/en/solve/examples/filtering-products-in-category/price-filter.java
final EvitaResponse<SealedEntity> entities = evita.queryCatalog(
	"evita",
	session -> {
		return session.querySealedEntity(
			query(
				collection("Product"),
				filterBy(
					entityLocaleEquals(Locale.forLanguageTag("en")),
					hierarchyWithin(
						"categories",
						attributeEquals("url", "/en/smartwatches")
					),
					priceInPriceLists("basic"),
					priceInCurrency(Currency.getInstance("EUR")),
					priceValidInNow(),
					userFilter(
						priceBetween(new BigDecimal("50"), new BigDecimal("400"))
					)
				),
				require(
					priceHistogram(10, STANDARD)
				)
			)
		);
	}
);
